tools/src/dbgen/Lib.hs

```diff
@@ -97,8 +99,19 @@ data AccountSpec = AccountSpec
 instance FromJSON AccountSpec
 instance ToJSON AccountSpec
 
-type AddressRange = Integer
-type DistributionAmount = Integer
+newtype AddressRange = AddressRange 
+    { unAddressRange :: Integer 
+    } deriving (Show, Eq, Generic)
+
+instance FromJSON AddressRange
+instance ToJSON AddressRange
+
+newtype DistributionAmount = DistributionAmount 
+    { unDistributionAmount :: Integer 
+    } deriving (Show, Eq, Generic)
+    
+instance FromJSON DistributionAmount
+instance ToJSON DistributionAmount
```

Comment:
okay, so this change indicates that we are serializing/deserializing these fields! That's very good to know. These changes are going to break the serialization format.

We should probably go back and ensure that if we're serializing/desererializing using `Generic`s, the `ToJSON` and `FromJSON` instances produce the same output as before.

For this instance, we'll want to write a test that has the result of calling `Data.Aeson.encode` on an old value of these types (that has the field accessors). Then we must ensure that the *new* serialization matches that exactly for a similar value.

One way we might do that is replicating the data type in the test suite exactly as it used to be, and then creating a function `delabel :: FakeUtxoCoinDistribution -> Real.FakeUtxoCoinDistribution` that converts. Then we write a QuickCheck property that asserts that `forAll fakeUtxoCoinDistribution $ \x -> encode (delabel x) === encode x`.